hack/gen-api-reference-docs.sh

```diff
@@ -160,9 +160,10 @@ main() {
         "${knative_eventing_sources_root}"
 
     log "Generated files written to ${clone_root}/out/."
-    log "Copy the files in reference/ directory to knative/docs."
-    if command -v open >/dev/null; then
```

Comment:
it'd be cool to still keep `command -v open` :) 

but I guess you're not merging this to master so it's cool.

Reply:
I could not get past this error: https://stackoverflow.com/questions/42463929/couldnt-find-a-file-descriptor-referring-to-the-console-on-ubuntu-bash-on-win

And the solution suggested worked. Not sure if you know of an alternate?

Also, while im thinking about these troubleshooting efforts, it reminded me that I ran into "no tmp space" storage size errors and had to go in an manually delete all the cloned content for each failed attempt of the command. I realized that the docs got generated just fine but since I reran the command until the "open command" error resolved, I ended up with 10-15 copies of all the clones. 

Perhaps an option to delete on error? Or some other magic you might have?

Reply:
> I could not get past this error: [stackoverflow.com/questions/42463929/couldnt-find-a-file-descriptor-referring-to-the-console-on-ubuntu-bash-on-win](https://stackoverflow.com/questions/42463929/couldnt-find-a-file-descriptor-referring-to-the-console-on-ubuntu-bash-on-win)

`open` is referring to the macOS tool here actually.  :) didn't realize linux has it too. dont worry about it here. in my new PR xdg comes first.

> it reminded me that I ran into "no tmp space" storage size errors and had to go in an manually delete all the cloned content for each failed attempt of the command.

this sounds odd and shouldn't happen (and doesn't happen).

> I realized that the docs got generated just fine but since I reran the command until the "open command" error resolved, I ended up with 10-15 copies of all the clones.

yeah that's a legit bug. but temp dirs are supposed to be cleaned up on every reboot. so I assumed people would not run low on disk. knative repos arent that big?

> Or some other magic you might have?

I think ideally we'd add a cleanup step running on errors etc. Opened https://github.com/knative/docs/issues/1000 to track.